lib/internal/Magento/Framework/Code/Reader/ClassReader.php

```diff
@@ -17,46 +21,62 @@ class ClassReader implements ClassReaderInterface
      *
      * @param  string $className
      * @return array|null
-     * @throws \ReflectionException
+     * @throws ReflectionException
      */
     public function getConstructor($className)
     {
-        $class = new \ReflectionClass($className);
+        $class = new ReflectionClass($className);
         $result = null;
         $constructor = $class->getConstructor();
         if ($constructor) {
             $result = [];
-            /** @var $parameter \ReflectionParameter */
+            /** @var $parameter ReflectionParameter */
             foreach ($constructor->getParameters() as $parameter) {
                 try {
                     $result[] = [
                         $parameter->getName(),
-                        $parameter->getClass() !== null ? $parameter->getClass()->getName() : null,
+                        $this->getParameterClass($parameter),
```

Comment:
This should evaluate to the string class name or null considering the previous implementation

Reply:
Changed according to previous implementation. 

Thanks!